src/grass_gis_helpers/tests.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:       lib with test related helper functions for GRASS GIS
+#
+# AUTHOR(S):    Anika Weinmann
+#
+# PURPOSE:      lib with test related helper functions for GRASS GIS
+#
+# COPYRIGHT:	(C) 2024 by mundialis and the GRASS Development Team
+#
+# 		This program is free software under the GNU General Public
+# 		License (>=v2). Read the file COPYING that comes with GRASS
+# 		for details.
+#
+#############################################################################
+
+import grass.script as grass
+
+
+def get_number_of_grass_elements():
+    """Get the number of grass elements like raster, vector and regions
+
+    Returns:
+        num_rast (int): The number of raster maps in current mapset
+        num_vect (int): The number of vector maps in current mapset
+        num_gr (int): The number of groups in current mapset
+        num_reg (int): The number of regions in current mapset
+        num_mapsets (int): The number of mapsets in current location
+    """
+    rast_list = grass.parse_command("g.list", type="raster")
+    vect_list = grass.parse_command("g.list", type="vector")
+    gr_list = grass.parse_command("g.list", type="group")
+    reg_list = grass.parse_command("g.list", type="region")
+    mapset_list = grass.parse_command("g.mapsets", flags="l", quiet=True)
+    num_rast = len(rast_list)
+    num_vect = len(vect_list)
+    num_gr = len(gr_list)
+    num_reg = len(reg_list)
+    num_mapsets = len(mapset_list)
+    return num_rast, num_vect, num_gr, num_reg, num_mapsets
+
+
+def check_number_of_grass_elements(
+    ref_num_rast, ref_num_vect, ref_num_gr, ref_num_reg, ref_num_mapsets
```

Comment:
Maybe it would be nice to have it optional if all should be tested or just specific given e.g. only raster

Reply:
I actually prefer that you have to test them all. By using it for the `r.dem.import` addon I see that there are some aoi vector maps which I have not expected. With this check I could add them to the cleanup function.